code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -64,21 +64,22 @@
 	hide(hides_under_flooring() && !T.is_plating())
 
 /obj/item/radio/beacon/anchored/attackby(obj/item/I, mob/living/user)
-	..()
 	if(istype(I, /obj/item/stack/nanopaste))
```

Comment:
maybe add check that it's non-functioning in first place? it'd be possible to just spam and waste nanopaste on a working radio otherwise.